torch/_dynamo/eval_frame.py

```diff
@@ -643,6 +645,177 @@ def __bool__(self):
+            if constraint_range is None:
+                continue
+
+            min_int_val = _convert_to_int(constraint_range.vr.lower)
+            max_int_val = _convert_to_int(constraint_range.vr.upper)
+
+            if min_int_val is None and max_int_val is None:
+                continue
+
+            dim = self.tracer.create_proxy(
+                "call_function", torch.ops.aten.sym_size, (arg, constraint.dim), {}
+            )
+            assert_msg = f"Input #{self.count}'s dimension #{constraint.dim} size is outside of supported dynamic range"
+
+            if min_int_val and min_int_val > 2:
+                gt = self.tracer.create_proxy(
+                    "call_function", operator.gt, (dim, min_int_val), {}
+                )
+                tensor_gt = self.tracer.create_proxy(
+                    "call_function", torch.scalar_tensor, (gt,), {}
+                )
+                self.tracer.create_proxy(
+                    "call_function",
+                    torch._assert_async,
+                    (tensor_gt, assert_msg),
+                    {},
+                )
+
+            if max_int_val and max_int_val < math.inf:
+                lt = self.tracer.create_proxy(
+                    "call_function", operator.lt, (dim, max_int_val), {}
+                )
+                tensor_lt = self.tracer.create_proxy(
+                    "call_function", torch.scalar_tensor, (lt,), {}
+                )
+                self.tracer.create_proxy(
+                    "call_function",
+                    torch._assert_async,
+                    (tensor_lt, assert_msg),
+                    {},
+                )
+
+        self.count += 1
+        return arg
+
+    def call_function(self, target, args, kwargs):
+        arg = super().call_function(target, args, kwargs)
+        from torch._export.constraints import constrain_as_size
+
+        if target == constrain_as_size:
+            min_val = kwargs["min"]
+            max_val = kwargs["max"]
+            gt = self.tracer.create_proxy(
+                "call_function", operator.gt, (args[0], min_val), {}
+            )
+            tensor_gt = self.tracer.create_proxy(
+                "call_function", torch.scalar_tensor, (gt,), {}
+            )
+            lt = self.tracer.create_proxy(
+                "call_function", operator.lt, (args[0], max_val), {}
+            )
+            tensor_lt = self.tracer.create_proxy(
+                "call_function", torch.scalar_tensor, (lt,), {}
+            )
+
+            self.tracer.create_proxy(
+                "call_function",
+                torch._assert_async,
+                (tensor_gt, "min_val"),
+                {},
+            )
+
+            self.tracer.create_proxy(
+                "call_function",
+                torch._assert_async,
+                (tensor_lt, "max_val"),
+                {},
+            )
+        return arg
+
+    def run_node(self, n):
```

Comment:
What does this override cover? just `output` nodes?

Reply:
This is to populate `val` and `tensor_dict` in node.meta